src/pages/index.js

```diff
@@ -17,51 +19,127 @@ import {
   profileEditButton,
 } from "../utils/constants.js";
 
-const cardSelector = "#card-template";
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "7ed3bb2f-7b5a-422d-a884-e2fa31c7bd62",
+    "Content-Type": "application/json",
+  },
+});
 
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   jobSelector: ".profile__description",
+  avatarSelector: ".profile__image",
 });
-const editProfilePopup = new PopupWithForm(
-  "#profile-edit-modal",
-  ({ name, description }) => handleEditProfileSubmit(name, description)
-);
 
-const addCardPopup = new PopupWithForm("#add-card-modal", ({ title, link }) => {
-  handleAddCardSubmit(title, link);
+const editAvatarPopup = new PopupWithForm("#modal-edit-avatar", (data) => {
+  return api
+    .updateAvatar(data)
+    .then((res) => {
+      userInfo.setUserInfo(res);
+    })
+    .catch((error) => {
+      console.error(error);
+    });
+});
+editAvatarPopup.setEventListeners();
+
+document
+  .querySelector("#profile-image-pencil")
+  .addEventListener("click", () => {
+    formValidators["edit-avatar-form"].resetValidation();
+    editAvatarPopup.open();
+  });
+
+const confirmAction = (card, cardId) => {
+  return api
+    .deleteCard(cardId)
+    .then(() => {
+      card.removeCard();
+    })
+    .catch((error) => {
+      console.error(error);
+    });
+};
+
+const deleteCardPopup = new PopupWithConfirmation({
+  popupSelector: "#delete-card-modal",
+  handleConfirm: confirmAction,
 });
+
+deleteCardPopup.setEventListeners();
+
 const iamgePreviewPopup = new PopupWithImage({
   popupSelector: "#preview-image-modal",
 });
 
-/* ----------------------- */
-/*      Event Listner      */
-/* ----------------------- */
-// eneble event listeners in each form
-editProfilePopup.setEventListeners();
-addCardPopup.setEventListeners();
-iamgePreviewPopup.setEventListeners();
-
-const renderCard = (cardData) => {
-  const newCard = new Card(
-    cardData,
-    cardSelector,
-    iamgePreviewPopup,
-    (title, link) => iamgePreviewPopup.open(title, link)
+let section;
+
+const renderCard = (data) => {
+  const cardElement = new Card(
+    {
+      data: data,
+      handleImageClick: (imageData) => {
+        iamgePreviewPopup.open(imageData);
+      },
+      handleDeleteCard: (card, cardId) => {
+        deleteCardPopup.open(card, cardId);
+      },
+      confirmPopup: deleteCardPopup,
+      api: api,
+    },
+    "#card-template"
   );
```

Comment:
It's highly recommended to make a separate function `createCard` for creating a card not to copy-paste the code

```js		

function createCard(item) {
    // here you create a card
  return cardElement.getView()
}

```



So, you'll have here

```js

const renderCard = (data) => {
  const cardElement = createCard(data)
  section.addItem(cardElement);
};
```